src/EventStore.Transport.Tcp/TcpConnection.cs

```diff
@@ -186,7 +186,7 @@ public class TcpConnection : TcpConnectionBase, ITcpConnection {
 			} else {
 				NotifySendCompleted(socketArgs.Count);
 
-				if (_closed != 0)
+				if (_isClosed)
```

Comment:
Do we not need to take a lock here?

Reply:
@hayley-jean I think that stale data is acceptable here based on what the previous code was doing. If `isClosed` is true but we miss it here we will get the value of `true` it in the next `TrySend()->ProcessSend()` call. Or alternatively, we will call `ReturnSendingSocketArgs()` in the `catch (ObjectDisposedException)` of `TrySend()`